### DISCUSSION TITLE
Tasmota DisplayText command: How to draw text over filled rectangle/circle with transparent background

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#21804] [Q&A] [answered]
I am trying to draw text over a filled circle as follows:

`DisplayText [z][Ci15Bi1][Ci16x199y199K40][Ci5x180y185f2s1]CAR`

1. Display background Color index is set to Black using `Ci15Bi1` command.
2. Blue Circle drawn using `[Ci16x199y199K40] `command.
3. Text drawn in Red color using `[Ci5x180y185f2s1]CAR` command.

The problem is the Text is drawn with black background over the blue filled circle.

I want the text to be drawn with a transparent background over the filled blue circle.

How should I do that?

Currently I get this:
![image](https://github.com/user-attachments/assets/c2b5f519-ac2f-480f-810c-05e9e891dc0e)

What I want is this:
![image](https://github.com/user-attachments/assets/00dd2a2a-ef6e-4646-b7f9-289b781ab43d)

**Comment by jkwim** [comment]
I managed to achieve the effect by blindly copying the following command from an example in the documentation.
`[dc19:31000]`
However there is no explanation as to what this command does.

Final command line is:
`DisplayText [z][Ci15Bi1][Ci16x199y199K40][dc19:31000][x180y185f2s1Ci5]CAR`

Can someone explain the command `[dc19:31000]` please?

**Reply by sfromis** [reply]
Not trying to explain, but I observe that the [docs](https://tasmota.github.io/docs/Displays/#displaytext) says:
> `dcI:V` = define index color entry Index 19-31, V 16 bit color value (index 0-18 is fixed)

And:
> You may expand the index color table up from index 19 to 31. the cmd [dcI:V] defines the index color with index I (19-31) to the 16 bit color value V

It looks like the color value V is in RGB565 representation

**Reply by jkwim** [reply]
So then the context is:

This is the code snippet I looked at 

```
; define index color
[dc19:31000]
; draw transparent text with new index color
[x60y30f2Ci19D2]Tasmota
```
First color index 19 is defined and then use that color with` Ci19` right?

Then the command for Transparent Text drawing must be the `D2` above.
`
DisplayText [Ci7x199y199K35][Ci0D2x175y180f2s1]CAR[x182y199f2s1]ON[D0][f1]
`
Based on a wild guess I used D2 to turn on transparent text but then all text started to become transparent on subsequent iterations.
So I made another wild guess and added D0 at the end to turn off and it seemed to work.

**Reply by jkwim** [reply, marked as answer]
Documentation had this but I could not comprehend it.

Dp = switch display drawing options:
bit 0: auto updates => 1 auto draw on each displaytext cmd, 0 display must be updated manually with d
( only valid for bw oled and epaper displays, color displays draw always immediately)
bit 1: character drawing => 0 opaque character drawing, 1 transparent character drawing

Now it makes sense.

D0 = 00
D1 = 01
D2 = 10
D3 = 11
